include/library/spdm_requester_lib.h

```diff
@@ -347,7 +351,9 @@ libspdm_return_t libspdm_start_session_ex(void *spdm_context, bool use_psk,
                                           void *requester_random,
                                           size_t *requester_random_size,
                                           void *responder_random,
-                                          size_t *responder_random_size);
+                                          size_t *responder_random_size,
+                                          void *opaque_data,
```

Comment:
Different from CHALLENGE and GET_MEASUREMENT, KEY_EXCHANGE and PSK_EXCHANGE have requester_opaque_data.

Do we want to allow integrator input requester_opaque_data ? or return requester_opaque_data from libspdm ?

Reply:
Probably should, but will need to integrate the Integrator's opaque data with DSP0277 opaque data. If `OpaqueDataFmt1` is selected then libspdm needs to know the number of opaque elements.

Reply:
I think we can just let integrator fill **everything**.

I think the logic could be:
```
if (requester_opaque_data == null) {
    opaque_key_exchange_req_size =
        libspdm_get_opaque_data_supported_version_data_size(spdm_context);
    libspdm_write_uint16(ptr, (uint16_t)opaque_key_exchange_req_size);
    ptr += sizeof(uint16_t);
    libspdm_build_opaque_data_supported_version_data(
        spdm_context, &opaque_key_exchange_req_size, ptr);
    ptr += opaque_key_exchange_req_size;
} else {
    opaque_key_exchange_req_size  = requester_opaque_data_size;
    libspdm_write_uint16(ptr, (uint16_t)opaque_key_exchange_req_size);
    ptr += sizeof(uint16_t);
    libspdm_copy_mem (ptr, requester_opaque_data_size, requester_opaque_data, requester_opaque_data_size)
}
```

We already did that in GET_CSR API.